server/misc_tracking.py

```diff
@@ -0,0 +1,222 @@
+import datetime
+import time
+from server.trackable import Trackable, create_trackable_from_row
+from flask import (
+    Blueprint, jsonify, request, session
+)
+
+from server.db import get_db
+
+bp = Blueprint('api', __name__, url_prefix='/api')
+date_format = '%Y-%m-%d'
+
+
+def get_user_by_api_key(request):
+    db = get_db()
+    api_key = request.json.get('api_key')
+    user = db.execute("SELECT * FROM user WHERE api_key=?",
+                      (api_key,)).fetchone()
+    if user is None:
+        return None
+    return user['id']
+
+
+@bp.route('/trackable-item', methods=['GET', 'POST'])
+def get_tracked_items():
+    db = get_db()
+
+    if request.method == 'POST':
+        user_id = get_user_by_api_key(request)
+        if user_id is None:
+            return jsonify({"data": []})
+    else:
+        user_id = session.get('user_id')
+
+    tracked_items = db.execute(
+        "SELECT * FROM tracked_items WHERE owner_id=?", (user_id,)).fetchall()
+
+    results = []
+    for t in tracked_items:
+        item = create_trackable_from_row(t).to_json()
+        results.append(item)
+
+    return jsonify(results)
+
+
+@bp.route('/trackable-item', methods=['PUT'])
+def create_trackable_item():
+    db = get_db()
+    cursor = db.cursor()
+    user_id = get_user_by_api_key(request)
+
+    if not request.is_json:
+        return jsonify({"message": "Content type is not supported."}), 415
+
+    data = request.json
+    name = data.get('name')
+    type = data.get('type')
+
+    cursor.execute(
+        "INSERT INTO tracked_items (owner_id, name, type) VALUES (?, ?, ?)",
+        (user_id, name, type))
+    db.commit()
+    return jsonify({"id": cursor.lastrowid})
+
+
+@bp.route('/trackable-item', methods=['DELETE'])
+def delete_trackable_item():
+    db = get_db()
+    user_id = session.get('user_id')
+
+    data = request.json
+    id = data.get('id')
+
+    if not request.is_json:
+        return jsonify({"message": "Content type is not supported."}), 415
+
+    deleted_item = db.execute(
+        "DELETE FROM tracked_items WHERE id = ? AND owner_id = ?", (id, user_id))
+    db.commit()
+    return jsonify(deleted_item)
+
+
+@staticmethod
+def validate_date(date, param):
+    try:
+        if date is not None:
+            datetime.datetime.strptime(date, date_format)
```

Comment:
Use python-dateutil library parse method which is does an outstanding job of doing the Right Thing when you throw random date strings at it